internal/appsec/waf/waf.go

```diff
@@ -544,6 +540,14 @@ func (e *encoder) encodeStruct(v reflect.Value, wo *wafObject, depth int) error
 			continue
 		}
 
+		// Use the json tag name as field name if present
+		if tag, ok := field.Tag.Lookup("json"); ok {
+			name := strings.Split(tag, ",")[0]
+			if len(name) > 0 {
+				fieldName = name
```

Comment:
Hotpath police: this is going to happen a lot in our world of web services where JSON is still the top most used format. So let's optimize this with `IndexByte()` instead:
```suggestion
			if i := strings.IndexByte(tag, byte(',')); i > 0 {
				tag = tag[:i]
			} 
			if len(tag) > 0 {
				fieldName = tag
```